code/__HELPERS/logging.dm

```diff
@@ -36,6 +36,22 @@
 	GLOB.STUI.admin.Add("\[[time_stamp()]]ADMIN: [text]")
 	GLOB.STUI.processing |= STUI_LOG_ADMIN
 
+/proc/check_slur(var/text, var/mob/source, var/method = "Speech")
+	if(!CONFIG_GET(flag/log_slurs))
+		return FALSE
+	var/list/TriggerWords = file2list("config/banned_words.txt")
```

Comment:
this should be cached so it doesn't read file2list all the time, and additionally be a regex. i gave it some thought and the best way to handle the scunthorpe problem would be to make the regex check for whitespace, common punctuation, and line ends surrounding the word. that way it won't trigger for everything and also be more performant than looping for a list

Reply:
additionally, instead of being "banned_words.txt", call it "slur_words.txt" for clarity, since we might want to ban other words in the near future winkwinknudgenudge

Reply:
Regex I don't know how to integrate into DM.
File renamed.
As for applying to other inputs I'll do so later when I've time :)

Reply:
@realforest2001 https://www.byond.com/docs/ref/#/proc/regex